servers/express.js

```diff
@@ -1,56 +1,139 @@
 'use strict';
 
+const fsp = require('fs/promises');
+const crypto = require('crypto');
 const path = require('path');
+
+const bodyParser = require('body-parser');
 const express = require('express');
+const multer = require('multer');
 
 const Skeleton = require('./skeleton');
 const watcher = require('../file-watcher.js');
 
+// multer configs
+const storage = multer.diskStorage({
+  destination: './uploads/',
+  filename: function(req, file, cb) {
+    const basename = crypto.randomBytes(16).toString('hex');
+    const ext = path.extname(file.originalname);
+    cb(null, `${basename}${ext}`);
+  }
+});
+const upload = multer({storage: storage});
+
+// minimist configs
 const argvOptions = {
   alias: {
-    l: 'logfile',
+    l: 'logfile'
   },
   default: {
-    l: 'route-metrics.log',
+    l: ''
   },
 };
 const argv = require('minimist')(process.argv.slice(2), argvOptions);
 
-// the datarows
+// datarows and globals
+let firstTs, lastTs, version;
 const eventloopDataRows = [];
 const memoryDataRows = [];
 const cpuDataRows = [];
-let version;
 
 // the app
-const pathToLogFile = argv.logfile;
+let pathToLogFile = argv.logfile;
+let uploadedFiles = [];
 const app = express();
 
-app.all('/*', function (req, res, next) {
+// create the routers
+const clientRoutes = express.Router();
+const apiRoutes = express.Router();
+
+app.all('/*', function(req, res, next) {
   res.header('Access-Control-Allow-Origin', '*');
   res.header('Access-Control-Allow-Headers', 'X-Requested-With');
   next();
 });
 
-app.use(express.static(path.join(__dirname, '..', 'front-end', 'build')));
+app.use(express.static(path.join(__dirname, '../front-end', 'build')));
+app.use(bodyParser.json());
+app.use('/api', apiRoutes);
+app.use('/', clientRoutes);
+
+apiRoutes.post('/logfiles', upload.any(), async(req, res) => {
+  for (const file of req.files) {
```

Comment:
i'm ok with removing uploading files altogether but, if we're going to upload even one file, then we need to handle multiple. if we only allow one file to be uploaded then there is practically no benefit over just restarting the server with a different filename.

the original implementation was simple by design: start a server with the file that should be displayed. once upload was added then it opens up scenarios where people might collect multiple files and access the server from multiple browser windows so they can compare different log files side-by-side. and if they change the time-slice or the visible time-series for a window, the data still needs to be there. i probably should have objected more strongly to the idea of adding upload than i did because it adds a significant level of complexity.